### ISSUE TITLE
CancelCommandTest is failing on ubuntu workflow

### ISSUE THREAD

**Issue by jan-gogo** [Flank/flank#1403]
**Describe the bug**
> A clear and concise description of what the bug is.

`CancelCommandTest` is failing on the ubuntu workflow because of missing `results/a/matrix_ids.json`.

**To Reproduce**
> Steps to reproduce the behavior:

CI: 
1. Create a pull request
1. Expect failed test in ubuntu workflow like that https://github.com/Flank/flank/runs/1562332906#step:6:260

Locally:
1. Clear the `results` directory
1. Run only `CancelCommandTest.cancelCommandRuns`
1. Expect failure like that https://github.com/Flank/flank/runs/1562332906#step:6:260

**Expected behavior**
> A clear and concise description of what you expected to happen.

The test should create `matrix_ids.json` on-demand before running `CancelCommand`.

**Additional context**
> Add any other context about the problem here.

This test is flaky in some situations. Can pass if `results/a/matrix_ids.json` already exists, possibly created by another test before.

**Comment by adamfilipow92** [comment]
If ```MatrixUtilTest.`should newly created directory and if do not use localResultDir and runPath not exist` ``` creates directory a/b with newest modify date. Function ```IArgs.getLastGcsPath()``` thinks this is the latest result dir and try to load ```matrix_ids.json``` from this directory. This directory doesn't contain this file so the test fails. 😄

**Comment by Sloox** [comment]
@adamfilipow92 a simple changes to not fail dramatically if the file is not found should fix it?

**Comment by pawelpasterz** [comment]
I think we should review the rest of the test code and use https://junit.org/junit4/javadoc/4.12/org/junit/rules/TemporaryFolder.html where ever it is possible (and makes sense)

**Comment by Sloox** [comment]
@pawelpasterz I am afailing to see if that will achieve anything for this test.

**Comment by adamfilipow92** [comment]
> @adamfilipow92 a simple changes to not fail dramatically if the file is not found should fix it?

I think we could remove first ```CancelCommand().run()``` to make this test independent of the rest. In the next step, we use ```AndroidRunCommand``` this command creates its own results dir so everything should works.

**Comment by Sloox** [comment]
> should wo

Nice! Lets go with this.

**Comment by pawelpasterz** [comment]
> @pawelpasterz I am afailing to see if that will achieve anything for this test.

I didn't say it will fix this issue itself. It was more about having self-cleaning-up tests. If UT left something after run it's generally bad idea. So in parallel with the fix, we can enhance our tests

**Comment by adamfilipow92** [comment]
> > @pawelpasterz I am afailing to see if that will achieve anything for this test.
> 
> I didn't say it will fix this issue itself. It was more about having self-cleaning-up tests. If UT left something after run it's generally bad idea. So in parallel with the fix, we can enhance our tests

Agree

**Comment by pawelpasterz** [comment]
But yeah, I think you are right. Let's have it solved in separate PR
